From 9712303fd79be5ca3a0a4fcf702e0e7a01730a89 Mon Sep 17 00:00:00 2001
Message-ID: <9712303fd79be5ca3a0a4fcf702e0e7a01730a89.1707833700.git.sharlatanus@gmail.com>
In-Reply-To: <9c3d89f5eb0359dc6493dfc0d24667d2a5c6b484.1707833700.git.sharlatanus@gmail.com>
References: <9c3d89f5eb0359dc6493dfc0d24667d2a5c6b484.1707833700.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 13 Feb 2024 11:03:21 +0000
Subject: [PATCH v4 8/8] gnu: python-myst-parser: Relax some requirements.

* gnu/packages/sphinx.scm (python-myst-parser) [arguments]
<#:test-flags>: Disable one more failing test.
<#:phases>: Add 'relax-requirements phase.

Change-Id: I4f77ce81e9f69b81b18f3faa546c7d8cd77adf50
---
 gnu/packages/sphinx.scm | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/gnu/packages/sphinx.scm b/gnu/packages/sphinx.scm
index e280c06f2b..12dee3df27 100644
--- a/gnu/packages/sphinx.scm
+++ b/gnu/packages/sphinx.scm
@@ -803,6 +803,7 @@ (define-public python-mpl-sphinx-theme
 (define-public python-myst-parser
   (package
     (name "python-myst-parser")
+    ;; The latest version (v2.0.0) require Sphinx >= v6.
     (version "0.18.1")
     (source (origin
               (method git-fetch)        ;for tests
@@ -821,7 +822,17 @@ (define-public python-myst-parser
      (list #:test-flags #~(list "-k" (string-append
                                       "not test_basic "
                                       "and not test_gettext_html "
-                                      "and not test_fieldlist_extension"))))
+                                      "and not test_fieldlist_extension "
+                                      "and not test_syntax_extensions"))
+           #:phases
+           #~(modify-phases %standard-phases
+               (add-after 'unpack 'relax-requirements
+                 (lambda _
+                   (substitute* "pyproject.toml"
+                     ;; "mdit-py-plugins~=0.3.1"
+                     (("0.3.1") "0.4.0")
+                     ;; "markdown-it-py>=1.0.0,<3.0.0"
+                     (("3.0.0") "4.0.0")))))))
     (native-inputs
      (list python-beautifulsoup4
            python-docutils
-- 
2.41.0

